gradle-jdks/src/test/groovy/com/palantir/gradle/jdks/GradleJdkPatcherIntegrationTest.groovy

```diff
@@ -0,0 +1,201 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.jdks
+
+import com.google.common.base.Splitter
+import com.google.common.collect.Iterables
+import com.palantir.gradle.jdks.setup.CommandRunner
+import nebula.test.IntegrationSpec
+import spock.lang.TempDir
+
+import java.nio.file.Files
+import java.nio.file.Path
+
+class GradleJdkPatcherIntegrationTest extends IntegrationSpec {
+
+    private static String GRADLE_7VERSION = "7.6.2";
+    private static String GRADLE_8VERSION = "8.4";
+    private static AmazonCorrettoJdkDistribution CORRETTO_JDK_DISTRIBUTION = new AmazonCorrettoJdkDistribution();
+    private static String CORRETTO_DISTRIBUTION_URL_ENV = "CORRETTO_DISTRIBUTION_URL";
+    private static String JDK_17_VERSION = "17.0.9.8.1";
+    private static String EXPECTED_JDK_LOG = "JVM:          17.0.9 (Amazon.com Inc. 17.0.9+8-LTS)";
+    private static String AMAZON_ROOT_CA_1_SERIAL = "143266978916655856878034712317230054538369994"
+    private static String AMAZON_CERT_ALIAS = "AmazonRootCA1Test";
+    private static String PALANTIR_3RD_GEN_SERIAL = "18126334688741185161";
+    private static String PALANTIR_3RD_GEN_ALIAS = "Palantir3rdGenRootCa";
+
+    @TempDir
+    private Path workingDir
+
+    def setup() {
+
+        // language=groovy
+        buildFile << """
+            buildscript {
+                repositories {
+                    mavenCentral()
+                }
+                dependencies {
+                    classpath files(FILES)
```

Comment:
Why do we need to do this? I think this is (more-or-less) what nebula does?

Reply:
Without this classpath addition I will be getting `plugin com.palantir.jdks` not found when running the `./gradlew command` (https://github.com/palantir/gradle-jdks/pull/338/files#diff-98ba650ac01c560cdf5ffae45e4a36b0f4786c3956183b2d9cc1d9759b3061ceR222)
When running `./gradlew` directly the classpath  is not injected.

Reply:
Ah, makes sense, as we not actually using nebula here.